File: AiSD/sem_3/lab_0/3/3_a.cpp
#include <iostream>
#include <vector>
#include <fstream>
#include <string>

using namespace std;

const int MAXN = 100;

int sx, sy, ex, ey; // начальная и конечная клетки
bool visited[MAXN][MAXN];
int len_shortestPath = -1;










int main() {

    string line;
    vector<vector<int>> maze;
    ifstream file("maze.txt");

    

    while (file) {

        vector<int> tmp;

        getline(file, line);
        //cout << line << endl;
        for (char c: line) {
            tmp.push_back(c - '0');
        }

        maze.push_back(tmp);
    }
    maze.pop_back();



    


    
    


    return 0;
}

File: AiSD/sem_3/lab_0/3/3_a.py

def is_valid_move(x, y, maze):
    # Проверяем, что координаты (x, y) находятся в пределах лабиринта и клетка свободна
    return 0 <= x < len(maze) and 0 <= y < len(maze[0]) and maze[x][y] == 0





def main():
    # Чтение лабиринта из файла
    with open('maze.txt', 'r') as file:
        maze = [list(map(int, line.strip())) for line in file]

    for x in maze:
        print(x)
                

   



if __name__ == "__main__":
    main()